lib/tasks/multitenant/tenants.rake

```diff
@@ -98,5 +132,25 @@ namespace :multitenant do
     def condition_update_tenant_id(time_start, time_end)
       proc { |object| (object.tenant_id == nil) | ((object.created_at >= Time.strptime(time_start, '%m/%d/%Y %H:%M %Z')) & (object.created_at <= Time.strptime(time_end, '%m/%d/%Y %H:%M %Z'))) }
     end
+
+    def scheduled_or_running_background_deletions
+      [
+        # I was thinking that future schedules shouldn't count towards concurrency
+        # *Sidekiq::ScheduledSet.new.select { job_is_a_background_deletion? _1 },
+        *Sidekiq::Queue.new("deletion").select { job_is_a_background_deletion? _1 },
```

Comment:
Isn't it kind of duplicated logic?
- fetching from queue "deletion"
- checking if `job.queue == "deletion"

Reply:
The thing is that originally there were more checks. Then decided to reduce them to just the target queue. But if the method is not used, then the logic to select the jobs may change in the future leaving behind items in the queue.

I didn't think it was a big issue.

But because you rise the point, I think to just simplify this and do the checks all within `scheduled_or_running_background_deletions` to keep it in one place and potentially avoid confusion in the future.